catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/Query.java

```diff
@@ -72,4 +72,12 @@
    * @return max time to wait for query results in milliseconds, 0 if no timeout should occur.
    */
   public long getTimeoutMillis();
+
+  /**
+   * Creates a new instance of the query with the same properties but a new filter
+   *
+   * @param newFilter Filter to be associated with the new query
+   * @return the new query object
+   */
+  public Query newInstanceWithFilter(Filter newFilter);
```

Comment:
❗️ Adding this abstract method can break API binary compatibility for clients if they implement their own implementation of `Query` .

https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_API_Interfaces

This interface is whitelisted.

https://github.com/codice/ddf/blame/master/distribution/docs/src/main/resources/content/_reference/_whitelists/catalog-whitelist.adoc#L86

We can create a new factory interface that is not whitelisted that only we would implement.

Reply:
With a factory, how do you determine which type of query to instantiate? For example, in DDF we have QueryImpl and OpensearchQuery, but in downstream projects we may have more. Is it safe for the factory to always return a QueryImpl?

Reply:
Or is the solution to have a service in DDF and then disable and override it in downstream projects?

Reply:
I think this concept is needed, we have had tons of issues related to this where we didn't include a sort order, or a count parameter.  But we can get probably everything we need w/o touching the interface.  

We can probably just add a QueryImpl(Filter filter) constructor to handle this behavior, and not mess with any interface or factory at all.  So this is essentially just a QueryImpl capability.

Reply:
The default factory would only help with `QueryImpl`.  You could use service ranking to have a different factory service become the system default.  But that still would not help with creating a `OpensearchQuery` unless there was a way to tell from the factory input that you needed to construct an opensearch query.

I agree with @millerw8 that a new constructor on `QueryImpl` would be the easiest step forward without impacting the API white list binary compatibility.